### PR TITLE
Feature/status command implementation

### PR DESCRIPTION
**Pull request by weaselix**

### CODE REVIEW

File: golemcli/src/context.rs [hunk 1]
```diff
@@ -249,6 +249,10 @@ impl CliCtx {
         }
         enabled
     }
+
+    pub fn get_golem_lock_path(&self) -> PathBuf {
+        self.data_dir.join(PathBuf::from("rinkeby").join("LOCK"))
```
Comment (tworec) [inline golemcli/src/context.rs]: why do you assume we are on **rinkeby**?

File: golemcli/src/commands/status.rs [hunk 2]
```diff
@@ -0,0 +1,603 @@
...
+        let connection_status = if self.net_status.is_connected {
+            String::from("ONLINE")
+        } else {
+            String::from("OFFLINE")
+        };
+
+        section_builder
+            .end_section()
+            .entry("Golem version", &self.running_status.golem_version)
+            .entry("Node name", &self.running_status.node_name)
```
Comment (tworec) [inline golemcli/src/commands/status.rs]: I'd try to format output similarily to `settings show`

```
┌───────────────────────────────────────────────────────────────────────┬─────────────┬─────────────────────────┐
│  description [name]                                                   │  value      │  type                   │
├───────────────────────────────────────────────────────────────────────┼─────────────┼─────────────────────────┤
│                                                                       │             │                         │
│  General                                                              │             │                         │
│                                                                       │             │                         │
│  Node name [node_name]                                                │  2rec iMac  │  str                    │
```

it looks much more readable and clear than

```
General
  Process State: running
  Components Status:
    docker: Docker is available
    hyperdrive: Connected to hyperg
    client: Connecting to network
    hypervisor: Docker VM is available
  Golem version: 0.19.2+dev603.g64869c2
  Node name: 2rec iMac
```

File: golemcli/src/commands/status.rs [hunk 3]
```diff
@@ -0,0 +1,603 @@
...
+                "Distributed files",
+                &self.running_status.disk_usage.distributed_files,
+            )
+            .end_section()
+            .end_section()
+            .new_section("Network")
+            .entry("Connection", &connection_status)
+            .entry(
+                "Port statuses",
+                &format!("{:?}", self.net_status.port_status),
```
Comment (tworec) [inline golemcli/src/commands/status.rs]: this should be more human readable, not json

File: golemcli/src/commands/status.rs [hunk 4]
```diff
@@ -0,0 +1,603 @@
...
+            )
+            .entry("Nodes online", &self.net_status.nodes_online.to_string())
+            .end_section()
+            .new_section("Account")
+            .entry("ETH address", &self.account_status.eth_address)
+            .entry("GNT available", &self.account_status.gnt_available)
+            .entry("ETH available", &self.account_status.eth_available)
```
Comment (tworec) [inline golemcli/src/commands/status.rs:573]:
```
  GNT available: 990.422335685398871004 GNT
  ETH available: 18.767548838 ETH
```
pls round this, maybe to 3 or 5 decimals

```
  GNT available: 990.422 GNT
  ETH available: 18.76754 ETH
```

**Review by tworec — CHANGES_REQUESTED** [review]
see my comments below. I'm also pushing minor improvements

File: golemcli/src/commands/status.rs (hunk 2 above)
Comment (weaselix) [reply golemcli/src/commands/status.rs]: I make it more readable without table. A table takes too much space.

File: golemcli/src/context.rs (hunk 1 above)
Comment (weaselix) [reply golemcli/src/context.rs]: Thanks, repaired.

File: golemcli/src/commands/status.rs (hunk 4 above)
Comment (weaselix) [reply golemcli/src/commands/status.rs:573]: It is not good idea to round this? What if someone has 0.000001 GNT?

File: golemcli/src/commands/status.rs (hunk 3 above)
Comment (weaselix) [reply golemcli/src/commands/status.rs]: done

File: golemcli/src/commands/status.rs (hunk 2 above)
Comment (tworec) [reply golemcli/src/commands/status.rs]: thx. I'd be great if you can show how it looks in tabular view, to compare both layouts

File: golemcli/src/commands/status.rs [hunk 5]
```diff
@@ -0,0 +1,644 @@
...
+                        account_status,
+                        provider_status,
+                        requestor_tasks_progress,
+                    }))
+                },
+            );
+        status
+    }
+
+    fn check_is_golem_run(is_mainnet: bool, ctx: CliCtx) -> bool {
```
Comment (tworec) [inline golemcli/src/commands/status.rs:271]:
```suggestion
    fn check_is_golem_running(is_mainnet: bool, ctx: CliCtx) -> bool {
```

File: golemcli/src/commands/status.rs [hunk 6]
```diff
@@ -0,0 +1,644 @@
...
+        let node_info = endpoint.as_golem_net().get_node().from_err();
+        let version = endpoint.as_golem().get_version().from_err();
+        let disk_usage = endpoint.as_golem_res().get_res_dirs_sizes().from_err();
+        let ctx_cloned = ctx.clone();
+
+        is_mainnet
+            .join5(server_status, node_info, version, disk_usage)
+            .map(
+                move |(is_mainnet, server_status, node_info, version, disk_usage)| {
+                    let is_golem_running = Section::check_is_golem_run(is_mainnet, ctx_cloned);
```
Comment (tworec) [inline golemcli/src/commands/status.rs:307]:
```suggestion
                    let is_golem_running = Section::check_is_golem_running(is_mainnet, ctx_cloned);
```

**Review by tworec — COMMENTED** [review]
you have build failed + pls rename fn mentioned below

**Comment by tworec** [comment]
pls check it with current `develop` as it yields
```
$ cargo run status
    Finished dev [unoptimized + debuginfo] target(s) in 0.15s
     Running `target/debug/golemcli status`
ERROR [golemcli] wamp.error.runtime_error: Should be a string of length <= 128
```

**Comment by weaselix** [comment]
> pls check it with current `develop` as it yields
> 
> ```
> $ cargo run status
>     Finished dev [unoptimized + debuginfo] target(s) in 0.15s
>      Running `target/debug/golemcli status`
> ERROR [golemcli] wamp.error.runtime_error: Should be a string of length <= 128
> ```
@tworec, which one Golem revision do you test against? I've tested it with the newest develop (bda35be9006b6cdd85201847e9ed985eacca0af8)- it works.

**Comment by tworec** [comment]
I'm using the same commit (bda35be90)


```
$ cargo run status
    Finished dev [unoptimized + debuginfo] target(s) in 0.49s
     Running `target/debug/golemcli status`
ERROR [golemcli] wamp.error.runtime_error: 'TaskState' object has no attribute 'status_message'
```

and on golem side

```
ERROR    [twisted                            ] AttributeError: 'TaskState' object has no attribute 'status_message': Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/autobahn/twisted/websocket.py", line 162, in _onMessage
    self.onMessage(payload, isBinary)
  File "/usr/local/lib/python3.6/site-packages/autobahn/wamp/websocket.py", line 95, in onMessage
    self._session.onMessage(msg)
  File "/usr/local/lib/python3.6/site-packages/autobahn/wamp/protocol.py", line 895, in onMessage
    on_reply = txaio.as_future(endpoint.fn, *invoke_args, **invoke_kwargs)
  File "/usr/local/lib/python3.6/site-packages/txaio/tx.py", line 429, in as_future
    return maybeDeferred(fun, *args, **kwargs)
--- <exception caught here> ---
  File "/usr/local/lib/python3.6/site-packages/twisted/internet/defer.py", line 150, in maybeDeferred
    result = f(*args, **kw)
  File "/Users/tworec/git/golem/golem/client.py", line 873, in get_tasks
    return list(filter(None, tasks))
  File "/Users/tworec/git/golem/golem/client.py", line 871, in <genexpr>
    tasks = (self.get_task(task_id) for task_id in task_ids)
  File "/Users/tworec/git/golem/golem/client.py", line 822, in get_task
    task_dict = self.task_server.task_manager.get_task_dict(task_id)
  File "/Users/tworec/git/golem/golem/task/taskmanager.py", line 1106, in get_task_dict
    state.to_dictionary(),
  File "/Users/tworec/git/golem/golem/task/taskstate.py", line 56, in to_dictionary
    'status_message': self.status_message,
builtins.AttributeError: 'TaskState' object has no attribute 'status_message'

```

**Comment by tworec** [comment]
The status look from branch [`feature/status_command_implementation_view_table`](https://github.com/golemfactory/golem-client/tree/feature/status_command_implementation_view_table) is much more elegant and readable. Go for it

```
                                         
 General                                 
                                         
 Process State:                         running 
 Components Status:                      
   docker:                              Docker is available 
   hyperdrive:                          Connected to hyperg 
   client:                              Connecting to network 
   hypervisor:                          Docker VM restarted 
 Golem version:                         0.20.0 
 Node name:                              
 Network:                               testnet 
 Disk usage:                             
   Received files:                      37M 
   Distributed files:                   37M 
                                         
 Network                                 
                                         
 Connection:                            ONLINE 
 Port statuses:                          
   40103:                               open 
   3282:                                open 
   40102:                               open 
 Nodes online:                          10 
                                         
 Account                                 
                                         
 ETH address:                           0x53c3B0311c38a6D435722175814390FC0C5968d9 
 GNT available:                         0.00 GNT 
 ETH available:                         0.00991 ETH 
                                         
 Provider Status                         
                                         
 Subtasks accepted (in session):        1 
 Subtasks rejected (in session):        0 
 Subtasks failed (in session):          0 
 Subtasks computed (in session):        1 
 Tasks in network (in session):         0 
 Pending payments:                      19995485493539192 
 Provider state:                        Idle 
                                         
 Requestor status                        
                                         
 all computed subtasks / all subtasks   0/0 
 [from all tasks]:                       
```

**Review by prekucki — APPROVED** [review]

**Pull request merged**